File: CinemaFrontend/src/app/services/film.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Film {
  id?: number;
  titre: string;
  description: string;
  genre: string;
  dureeMinutes: number;
  ageMinimum: number;
}

@Injectable({
  providedIn: 'root'
})
export class FilmService {
  private apiUrl = 'http://localhost:5062/api/films';

  constructor(private http: HttpClient) { }

  getFilms(): Observable<Film[]> {
    return this.http.get<Film[]>(this.apiUrl);
  }

  addFilm(film: Film): Observable<Film> {
    return this.http.post<Film>(this.apiUrl, film);
  }
}

File: CinemaBackend/Models/Film.cs
namespace CinemaBackend.Models;

public class Film
{
    public int Id { get; set; }
    public string Titre { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string Genre { get; set; } = string.Empty;
    public int DureeMinutes { get; set; }
    public int AgeMinimum { get; set; }
    public string? PosterPath { get; set; }
}